x/crosschain/types/cctx.go

```diff
@@ -82,10 +83,14 @@ func (m *CrossChainTx) AddRevertOutbound(gasLimit uint64) error {
 	if m.IsCurrentOutTxRevert() {
 		return fmt.Errorf("cannot revert a revert tx")
 	}
+	if len(m.OutboundTxParams) == 0 {
+		return fmt.Errorf("cannot revert before trying to process an outbound tx")
+	}
+
 	revertTxParams := &OutboundTxParams{
 		Receiver:           m.InboundTxParams.Sender,
 		ReceiverChainId:    m.InboundTxParams.SenderChainId,
-		Amount:             m.InboundTxParams.Amount,
+		Amount:             m.GetCurrentOutTxParam().Amount,
```

Comment:
@brewmaster012  made this change for the revert amount. I think the revert should be accounting for the first outbound we try. Let me know if you have any concerns

Reply:
I think this is a correct fix but I think this value has never been used because we use outbound[0].amount as a reference to pay the gas and this function overwrite the outbound[1].amount: 
https://github.com/zeta-chain/node/blob/1051a3115b3ffe565bc7677cc57b600a62080306/x/crosschain/keeper/processs_outbound.go#L80